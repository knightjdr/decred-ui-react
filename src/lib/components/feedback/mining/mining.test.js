import React from 'react';
import { shallow } from 'enzyme';

import Color from '../../../shared/color';
import Content from './mining-helper-content';
import Mining from './mining';
import Scale from '../../../shared/scale';
import Theme from '../../../shared/theme';

// mocks
jest.mock('../../../shared/color');
jest.mock('./mining-helper-content');

// mock scale
jest.mock('../../../shared/scale');
Scale.mockReturnValue({
  containerScale: 40,
});

// mock theme
jest.mock('../../../shared/theme');
Theme.mockReturnValue({
  accent1: '#000',
  accent2: '#000',
  primary: '#000',
});

describe('Mining', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('Renders with defaults', () => {
    Color.mockReturnValue(null);
    const wrapper = shallow(<Mining />);
    expect(wrapper).toMatchSnapshot();
    expect(Color).toHaveBeenCalledTimes(0);
    expect(Content).toHaveBeenCalledWith(
      '#000',
      '#000',
      '#000',
      40,
      1.6,
    );
    expect(Scale).toHaveBeenCalledTimes(1);
    expect(Scale).toHaveBeenCalledWith(1, { containerHeight: 40 });
    expect(Theme).toHaveBeenCalledTimes(1);
    expect(Theme).toHaveBeenCalledWith('primaryNegative', 'mining');
  });

  test('Top level container should have class matching componente', () => {
    Color.mockReturnValue(null);
    const wrapper = shallow(<Mining />);
    const container = wrapper.find('div').first();
    expect(container.hasClass('decred-ui-mining-container')).toBeTruthy();
  });

  test('Blockcolor changes color arg used for Content', () => {
    Color.mockReturnValue('#FD714A');
    shallow(<Mining blockColor="orange" />);
    expect(Color).toHaveBeenCalledTimes(1);
    expect(Content).toHaveBeenCalledWith(
      '#FD714A',
      '#000',
      '#000',
      40,
      1.6,
    );
  });

  test('Pickaxe changes color arg used for Content', () => {
    Color.mockReturnValue('#FD714A');
    shallow(<Mining pickAxeColor="orange" />);
    expect(Color).toHaveBeenCalledTimes(2);
    expect(Content).toHaveBeenCalledWith(
      '#000',
      '#FD714A',
      '#FD714A',
      40,
      1.6,
    );
  });
});
