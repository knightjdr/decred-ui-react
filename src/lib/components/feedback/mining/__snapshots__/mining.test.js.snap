// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mining Renders with defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mining
    blockColor={null}
    duration={1.6}
    pickAxeColor={null}
    scale={1}
    theme="primaryNegative"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "decred-ui-mining-container",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "decred-ui-mining-container",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
